experimental/ActionPolyRing/src/IO.jl

```diff
@@ -0,0 +1,209 @@
+###############################################################################
+#
+#  Action polynomial rings
+#
+###############################################################################
+
+### Difference ###
+function Base.show(io::IO, ::MIME"text/plain", dpr::DifferencePolyRing)
+  io = pretty(io)
+  n = nelementary_symbols(dpr)
+  print(io, "Difference polynomial ring in $n elementary symbols ")
+  for i in 1:n-1
+    print(io, string(elementary_symbols(dpr)[i]) * ", ")
+  end
+  print(io, string(elementary_symbols(dpr)[n])*"\n")
+  print(io, "with $(ndiffs(dpr)) commuting endomorphisms\n")
+  print(io, Indent())
+  print(io, "over $(base_ring(dpr))")
+  print(io, Dedent())
+end
+
+function Base.show(io::IO, dpr::DifferencePolyRing)
+  io = pretty(io)
+  if is_terse(io)
+    print(io, "Difference polynomial ring")
+  else
+    print(terse(io), "Difference polynomial ring in $(nelementary_symbols(dpr)) elementary symbols over ")
+    print(terse(io), base_ring(dpr))
+  end
+end
+
+### Difference ###
+function Base.show(io::IO, ::MIME"text/plain", dpr::DifferentialPolyRing)
+  io = pretty(io)
+  n = nelementary_symbols(dpr)
+  print(io, "Differential polynomial ring in $n elementary symbols ")
+  for i in 1:n-1
+    print(io, string(elementary_symbols(dpr)[i]) * ", ")
+  end
+  print(io, string(elementary_symbols(dpr)[n])*"\n")
+  print(io, "with $(ndiffs(dpr)) commuting derivations\n")
+  print(io, Indent())
+  print(io, "over $(base_ring(dpr))")
+  print(io, Dedent())
+end
+
+function Base.show(io::IO, dpr::DifferentialPolyRing)
+  io = pretty(io)
+  if is_terse(io)
+    print(io, "Differential polynomial ring")
+  else
+    print(terse(io), "Differential polynomial ring in $(nelementary_symbols(dpr)) elementary symbols over ")
+    print(terse(io), base_ring(dpr))
+  end
+end
+
+###############################################################################
+#
+# Expressify
+#
+###############################################################################
+
+@enable_all_show_via_expressify ActionPolyRingElem
+
+function _expressify_monomial!(prod::Expr, x, e)
+    @inbounds for i in 1:length(e)
+        if e[i] > 1
+            push!(prod.args, Expr(:call, :^, x[i], e[i]))  # deepcopy not needed for immutables
+        elseif e[i] == 1
+            push!(prod.args, x[i])
+        end
+    end
+end
+
+function expressify(a::ActionPolyRingElem, x = symbols(parent(a)); context = nothing)
+    sum = Expr(:call, :+)
+
+    for (c, e) in zip(coefficients(a), exponents(a))
+        if all(zero(eltype(e)) == ei for ei in e)  # constant term
+            push!(sum.args, expressify(c, context = context))
+        else
+            prod = Expr(:call, :*)
+            if !(isone(c))
+                push!(prod.args, expressify(c, context = context))
+            end
```

Comment:
```suggestion
            push!(prod.args, expressify(c, context = context))
```
you don't need to do any such checks here as `show_via_expressify` will do them for you